lib/swagger/generator.rb

```diff
@@ -276,22 +292,23 @@ def generate_swagger_paths(base_path=false, hide_custom_paths=false)
           end
         end
       end
-      if !hide_custom_paths
-        generate_swagger_custom_paths(custom_routes, base_path)
+      unless ignore_custom_actions
+        generate_swagger_custom_paths(custom_routes)
       end
     end
 
-    def generate_swagger_custom_paths(custom_routes, base_path)
-      name   = resource_name
-      plural = collection_name
+    def generate_swagger_custom_paths(custom_routes)
+      name   = swagger_resource_class_name || resource_name
+      plural = swagger_collection_name || collection_name
+      tags   = [ plural ]
 
       custom_routes.each do |route|
-        path = route[:path].gsub(base_path, '')
+        path = route[:path].gsub(swagger_base_path, '')
         if path.include?(':id')
           path = path.gsub(':id', '{id}')
           swagger_path "#{ path }" do
             operation route[:method] do
-              key :tags, [ plural ]
+              key :tags, tags
               key :summary, 'Custom method, Please Overwrite this Method Manually'
```

Comment:
This description is not clear. Use: 'Default action schema is used, custom Swagger description is required.'